app/src/main/kotlin/com/google/samples/apps/nowinandroid/ui/NiaAppState.kt

```diff
@@ -142,17 +142,22 @@ class NiaAppState(
     fun navigateToTopLevelDestination(topLevelDestination: TopLevelDestination) {
         trace("Navigation: ${topLevelDestination.name}") {
             val topLevelNavOptions = navOptions {
+                val startDestination = navController.graph.findStartDestination()
+
+                // Avoid popUpTo and navigate to same screen.
+                val shouldNotSavedAndRestored = startDestination.hasRoute(topLevelDestination.route)
```

Comment:
You could remove the double negative `!shouldNotSavedAndRestored` by renaming the property to `saveStateForThisDestination`. This would also make the code more legible.